google/cloud/bigtable/examples/bigtable_samples_instance_admin.cc

```diff
@@ -242,6 +285,231 @@ void DeleteCluster(google::cloud::bigtable::InstanceAdmin instance_admin,
+  bool instanceExists =
+      instances.end() !=
+      std::find_if(
+          instances.begin(), instances.end(),
+          [&instance_name](google::bigtable::admin::v2::Instance const& i) {
+            return i.name() == instance_name;
+          });
+  // [END bigtable_check_instance_exists]
+
+  // Create instance if does not exists
+  if (not instanceExists) {
+    // [START bigtable_create_prod_instance]
+    std::cout << "\nCreating a PRODUCTION Instance: ";
+    google::cloud::bigtable::DisplayName display_name("Put description here");
+
+    // production instance needs at least 3 nodes
+    auto cluster_config = google::cloud::bigtable::ClusterConfig(
+        zone_1, 3, google::cloud::bigtable::ClusterConfig::HDD);
+    google::cloud::bigtable::InstanceConfig config(
+        google::cloud::bigtable::InstanceId(instance_id), display_name,
+        {{cluster_id_1.get(), cluster_config}});
+    config.set_type(google::cloud::bigtable::InstanceConfig::PRODUCTION);
+
+    auto instance_details = instance_admin.CreateInstance(config).get();
+    std::cout << " Done" << std::endl;
+    // [END bigtable_create_prod_instance]
+  } else {
+    std::cout << "\nInstance " << instance_id.get() << " already exists."
+              << std::endl;
+    return;
+  }
+
+  // [START bigtable_list_instance]
+  std::cout << "\nListing Instances: " << std::endl;
+  auto instances_after = instance_admin.ListInstances();
+  for (auto const& instance : instances_after) {
+    std::cout << instance.name() << std::endl;
+  }
+  // [END bigtable_list_instance]
+
+  // [START bigtable_get_instance]
+  std::cout << "\nGet Instance: " << std::endl;
+  auto instance = instance_admin.GetInstance(instance_id.get());
+  std::string instance_detail;
+  google::protobuf::TextFormat::PrintToString(instance, &instance_detail);
+  std::cout << "GetInstance details : " << std::endl << instance_detail;
+  // [END bigtable_get_instance]
+
+  // [START bigtable_list_cluster]
+  std::cout << "\nListing Clusters: " << std::endl;
+  auto cluster_list = instance_admin.ListClusters(instance_id.get());
+  std::cout << "Cluster Name List: " << std::endl;
+  for (auto const& cluster : cluster_list) {
+    std::cout << "Cluster Name: " << cluster.name() << std::endl;
+  }
+  // [END bigtable_list_cluster]
+
+  // [START bigtable_create_cluster]
+  std::cout << "\nAdding Cluster: " << std::endl;
+  auto cluster_config = google::cloud::bigtable::ClusterConfig(
+      zone_2, 3, google::cloud::bigtable::ClusterConfig::HDD);
+  auto cluster =
+      instance_admin.CreateCluster(cluster_config, instance_id, cluster_id_2);
+  std::cout << "Cluster Created " << cluster.get().name() << std::endl;
+  // [END bigtable_create_cluster]
+
+  // [START bigtable_delete_cluster]
+  std::cout << "\nDeleting Cluster: ";
+  instance_admin.DeleteCluster(instance_id, cluster_id_2);
+  std::cout << " Done" << std::endl;
+  // [END bigtable_delete_cluster]
+
+  // [START bigtable_delete_instance]
```

Comment:
`bigtable_delete_instance` should be in a separate method like node.js

Reply:
fixed.